app/models/assessment_particulars.rb

```diff
@@ -0,0 +1,23 @@
+class AssessmentParticulars < RecursiveOpenStruct
+  def initialize(assessment)
+    super initial_data(assessment), recurse_over_arrays: true
+  end
+
+  private
+
+  def initial_data(assessment)
+    {
+      assessment_id: assessment.id,
+      client_reference_id: assessment.client_reference_id,
+      request: RecursiveOpenStruct.new(JSON.parse(assessment.request_payload), recurse_over_arrays: true),
+      response: RecursiveOpenStruct.new(initial_response_data, recurse_over_arrays: true)
+    }
```

Comment:
If `RecursiveOpenStruct` is recursive, why does it need to be call again on all of the recursions?

Would this have the same end result:

```ruby
    {
      assessment_id: assessment.id,
      client_reference_id: assessment.client_reference_id,
      request: JSON.parse(assessment.request_payload),
      response: initial_response_data
    }
```